src/main/java/org/example/enums/CourseType.java

```diff
@@ -0,0 +1,23 @@
+package org.example.enums;
+
+import org.example.exceptions.InvalidArgument;
+
+public enum CourseType {
+  A,
+  B,
+  C,
+  D,
+  E,
+  F;
+
+  public static CourseType fromString(String type) {
```

Comment:
function parameters should be final

Reply:
made function parameters final where ever parameters weren't reassigned